test/functional/test-standard-actions.js

```diff
@@ -419,6 +419,94 @@ describes.sandboxed('StandardActions', {}, () => {
       });
     });
 
+    describe('closeOrNavigateTo', () => {
+      let navigateToStub;
+      let closeOrNavigateToSpy;
+      let winCloseSpy;
+
+      beforeEach(() => {
+        navigateToStub = sandbox.stub(standardActions, 'handleNavigateTo')
+            .returns(Promise.resolve());
+
+        closeOrNavigateToSpy = sandbox.spy(standardActions,
+            'handleCloseOrNavigateTo');
+
+        win.close = () => {
+          win.closed = true;
+        };
+        winCloseSpy = sandbox.spy(win, 'close');
+
+        // Fake ActionInvocation.
+        invocation.method = 'closeOrNavigateTo';
+        invocation.args = {
+          url: 'http://bar.com',
+        };
+        invocation.caller = {tagName: 'DIV'};
+      });
+
+      it('should be implemented', () => {
```

Comment:
Ideally new tests would be written in async/await since they are easier to reason about.

```javascript
it('should be implemented', async() => {
  await standardActions.handleAmpTarget(invocation);
  expect(closeOrNavigateToSpy).to.be.calledOnce;
  expect(closeOrNavigateToSpy).to.be.calledWithExactly(invocation);
});
```